File: create-polos-py/src/create_polos/providers.py
from dataclasses import dataclass


@dataclass
class ProviderConfig:
    label: str
    value: str
    provider_string: str
    model_string: str
    pip_extra: str
    env_var: str
    env_placeholder: str


PROVIDERS: list[ProviderConfig] = [
    ProviderConfig(
        label="Anthropic",
        value="anthropic",
        provider_string="anthropic",
        model_string="claude-sonnet-4-5",
        pip_extra="anthropic",
        env_var="ANTHROPIC_API_KEY",
        env_placeholder="sk-ant-...",
    ),
    ProviderConfig(
        label="OpenAI",
        value="openai",
        provider_string="openai",
        model_string="gpt-4o-mini",
        pip_extra="openai",
        env_var="OPENAI_API_KEY",
        env_placeholder="sk-...",
    ),
    ProviderConfig(
        label="Google Gemini",
        value="google",
        provider_string="gemini",
        model_string="gemini-2.0-flash",
        pip_extra="gemini",
        env_var="GOOGLE_GENERATIVE_AI_API_KEY",
        env_placeholder="AIza...",
    ),
]

File: create-polos-ts/src/providers.ts
export interface ProviderConfig {
  label: string;
  value: string;
  import: string;
  call: string;
  package: string;
  packageVersion: string;
  envVar: string;
  envPlaceholder: string;
}

export const providers: ProviderConfig[] = [
  {
    label: 'Anthropic',
    value: 'anthropic',
    import: `import { anthropic } from '@ai-sdk/anthropic'`,
    call: `anthropic('claude-sonnet-4-5')`,
    package: '@ai-sdk/anthropic',
    packageVersion: '^3.0.0',
    envVar: 'ANTHROPIC_API_KEY',
    envPlaceholder: 'sk-ant-...',
  },
  {
    label: 'OpenAI',
    value: 'openai',
    import: `import { openai } from '@ai-sdk/openai'`,
    call: `openai('gpt-4o-mini')`,
    package: '@ai-sdk/openai',
    packageVersion: '^3.0.0',
    envVar: 'OPENAI_API_KEY',
    envPlaceholder: 'sk-...',
  },
  {
    label: 'Google Gemini',
    value: 'google',
    import: `import { google } from '@ai-sdk/google'`,
    call: `google('gemini-2.0-flash')`,
    package: '@ai-sdk/google',
    packageVersion: '^3.0.0',
    envVar: 'GOOGLE_GENERATIVE_AI_API_KEY',
    envPlaceholder: 'AIza...',
  },
];
